test/build.spec.ts

```diff
@@ -265,6 +265,11 @@ describe('Resolved project building', () => {
 				error: [reject],
 			};
 			const newTask = resolveTask(task, 'i386', 'qemux86', resolveListeners);
+
+			// also test that a `platform: undefined` value in `task.dockerOpts`
+			// does not override a valid value in `task.dockerPlatform`
+			task.dockerOpts.platform = undefined;
```

Comment:
With this modification to the test, the test would fail without this PR's code changes to `lib/build.ts`.